crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -54,28 +57,33 @@ pub struct OnSettlementEventUpdater {
     pub contract: GPv2Settlement,
     pub native_token: H160,
     pub db: Postgres,
-    pub current_block: CurrentBlockStream,
 }
 
 impl OnSettlementEventUpdater {
-    pub async fn run_forever(self) -> ! {
+    pub async fn run_forever(self, block_stream: CurrentBlockStream) -> ! {
+        let mut current_block = *block_stream.borrow();
+        let mut block_stream = into_stream(block_stream);
         loop {
-            match self.update().await {
-                Ok(true) => (),
-                Ok(false) => tokio::time::sleep(Duration::from_secs(10)).await,
+            match self.update(current_block.number).await {
+                Ok(performed_update) => {
+                    tracing::debug!(
+                        ?performed_update,
+                        block = current_block.number,
```

Comment:
I think logging the current block might be confusing. Intuitively I would assume it means "event of block X was processed" when it fact it means "event from this or an earlier block was processed and the current block happens to be X".